extensions/wikia/UserLogin/UserLoginSpecialController.class.php

```diff
@@ -82,14 +82,23 @@ public function index() {
 		$contentLangCode = $this->app->wg->ContLang->getCode();
 
 		// Redirect to standalone NewAuth page if extension enabled
-		if ( $this->app->wg->EnableNewAuthModal && $this->wg->request->getVal( 'type' ) !== 'forgotPassword' ) {
-			$newLoginPageUrl = '/signin?redirect=' . $this->userLoginHelper->getRedirectUrl();
+		if ( $this->app->wg->EnableNewAuthModal ) {
+			$isForgotPassword = $this->wg->request->getVal( 'type' ) == 'forgotPassword';
+			$redirectUrl = '/';
+
+			if ($isForgotPassword) {
+				$redirectUrl .= 'forgotpassword';
+			} else {
+				$redirectUrl .= 'signin';
+			}
```

Comment:
Extract the resource calculation?

`$redirectUrl = $this->getAuthenticationResource();`

```
private function getAuthenticationResource() {
    if ( $this->wg->request->getVal( 'type' ) == 'forgotPassword' ) {
        return '/forgotpassword';
    } else {
         return '/signin';
    }
}
```